Documentation/Coherency.md

```diff
@@ -0,0 +1,74 @@
+
+# Solutions Considered
+We have considered a few solutions:
+
+## Continue with traditional source-build coherence
+This would mean that we would continue to force coherence on every build, using our traditional method of passing around PackageVersions.props as the build progresses.  We would continue to use only one commit per repo, usually the latest on the relevant branch (e.g. release/3.0).  Currently we have to sometimes manually change commits if not all the repos' latest commits work together but if we're getting updates from BAR this problem should mostly go away.
+
+Pros:
+- Simple.
+- Already have this mechanism.
+
+Cons:
+- We continue to not ship bits that match official builds, except on releases.
+- We have to deal with the problem of not getting all packages for servicing releases:
+	-  Issues https://github.com/dotnet/source-build/issues/210 and https://github.com/dotnet/source-build/issues/923 are our earlier discussions.
+	-  If we turn on BuildAllConfigurations and BuildAllPackages, we will ship bits that do not exist except on RHEL machines.  If someone shares a project but not the packages directory, restoring these packages that do not officially exist will fail.
+	-  In the worst case, these packages that we force to build could be known-bad code (that is okay normally because that package is not shipping).
+	-  If we don't turn on BuildAllConfigurations and BuildAllPackages, we will continue to have a large number of prebuilt packages.
+
+## Build multiple versions of each repo
+This would mean we build CLI with its expected version of CoreFX, and core-setup with its own expected version of CoreFX.
+
+Pros:
+- We match the official bits very closely.
+- We don't have to deal with servicing considerations.
+
+Cons:
+- This explodes rather quickly and would have extremely long build times.
+- We have to be careful about cross-contaminating package feeds during the builds.
+
+## Store and download packages in later builds
+During each build, we would save source-built packages off in an Azure account or NuGet feed.  During later builds, we would restore these packages and use them in builds later in the dependency tree.  We can store the packages in multiple formats; nupkgs for us or RPM/Debs to allow distro maintainers to save the packages and reuse them.
+
+Pros:
+- Improved build times.
+	- Fringe benefit: incremental builds would probably finally work.
+- Still lets us eliminate prebuilt packages (all packages stored are source-built).
+- We match the official bits very closely.
+
+Cons:
+- Have to be very careful to not contaminate the storage with prebuilt packages.
+- Need internet access during the production dev and daily builds.
+- Need to carry required packages in a tarball or to an N+1 build (could be nupkg, tar.gz, RPM, deb).
+- Probably does not satisfy requirements for no prebuilt binaries.
```

Comment:
This point seems like it can be removed with RPM/Deb carryover.

Reply:
Removed.